src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int RATE_INDEX = 3;
+    private SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Map<String, Integer> employeeSalaries = new HashMap<>();
+        StringBuilder result = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo + System.lineSeparator());
+
+        try {
+            Date newDateFrom = parseDate(dateFrom);
+            Date newDateTo = parseDate(dateTo);
+
+            processSalaryData(data, employeeSalaries, newDateFrom, newDateTo);
+            buildSalaryReport(result, names, employeeSalaries);
+
+            return result.toString();
+        } catch (ParseException e) {
+            return "Caught ParseException";
+        }
+    }
+
+    private Date parseDate(String date) throws ParseException {
```

Comment:
[Use LocalDate instead of Date.](https://github.com/mate-academy/jv-salary-info/blob/master/checklist.md#use-localdate-instead-of-date)